sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SerializationDiagnosticsContext.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.ZonedDateTimeSerializer;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+public class SerializationDiagnosticsContext {
+    public volatile List<SerializationDiagnosticsContext.SerializationDiagnostics> serializationDiagnosticsList;
+
+    public void addMetaDataDiagnostic(SerializationDiagnosticsContext.SerializationDiagnostics serializationDiagnostics) {
+        if (serializationDiagnosticsList == null) {
+            serializationDiagnosticsList = new ArrayList<>();
+        }
+
+        serializationDiagnosticsList.add(serializationDiagnostics);
```

Comment:
I am guessing this can be accessed from multiple threads. Then shouldn't this be a synchronized list?

see this similar race condition bug in the diagnostics string: https://github.com/Azure/azure-sdk-for-java/pull/8505